stablehlo/dialect/TypeInference.cpp

```diff
@@ -946,9 +958,14 @@ LogicalResult inferSliceOp(Optional<Location> location, Value operand,
   SmallVector<int64_t, 4> limit(limitIndices.getValues<int64_t>());
   SmallVector<int64_t, 4> strideVals(strides.getValues<int64_t>());
 
+  llvm::SmallVector<int64_t> inputBounds;
+  BoundedDialectInterface* dialect = getBounds(rankedTy, inputBounds);
+
   SmallVector<int64_t, 4> shape;
   shape.reserve(rank);
+  SmallVector<int64_t> resultBounds;
   for (int64_t i = 0, e = rank; i != e; i++) {
+    if (!inputBounds.empty()) resultBounds.push_back(inputBounds[i]);
```

Comment:
this seems like too conservative? In the test we added, we know that the 2nd dim has size at most 2 with [0, 4) with strides = 2?

Good test!

Reply:
Thanks for bring up this!
Yes I'm thinking of this today as well. IIUC, the bound should play the role similar to dim_size, please see my new fixed commit.